test/runtests.sh

```diff
@@ -13,7 +13,10 @@ fi
 javac  ../src/seedu/addressbook/AddressBook.java -d ../bin
 
 # run the program, feed commands from input.txt file and redirect the output to the actual.txt
-java -classpath ../bin seedu.addressbook.AddressBook < input.txt > actual.txt
+java -classpath ../bin seedu.addressbook.AddressBook ' ' < /dev/null > actual.txt
+java -classpath ../bin seedu.addressbook.AddressBook 'directoryThatDoesNotExist/valid.filename' < /dev/null >> actual.txt
+java -classpath ../bin seedu.addressbook.AddressBook '.noFilename' < /dev/null >> actual.txt
+java -classpath ../bin seedu.addressbook.AddressBook < input.txt >> actual.txt
```

Comment:
Some additional test cases you may wish to add:

* explicit valid parent directory that exists, and a valid regular file that exists (e.g. `data/addressbook.txt`, where `data` is a directory that exists.)
* explicit valid parent directory that exists, and file name points to a not-a-regular-file (e.g. `data/addressbook.txt`, where `data/addressbook.txt` is a directory)

The 1st case can go something like this, I think:
```diff
diff --git a/test/exitinput.txt b/test/exitinput.txt
index e69de29..9a57223 100644
--- a/test/exitinput.txt
+++ b/test/exitinput.txt
@@ -0,0 +1,2 @@
+# exit directly
+exit
diff --git a/test/expected.txt b/test/expected.txt
index 3ea19a2..2109d91 100644
--- a/test/expected.txt
+++ b/test/expected.txt
@@ -30,6 +30,15 @@
 || AddessBook Level 1 - Version 1.0
 || Welcome to your Address Book!
 || ===================================================
+|| Enter command: || [Command entered:exit]
+|| Exiting Address Book... Good bye!
+|| ===================================================
+|| ===================================================
+|| ===================================================
+|| ===================================================
+|| AddessBook Level 1 - Version 1.0
+|| Welcome to your Address Book!
+|| ===================================================
 || Using default storage file : addressbook.txt
 || Enter command: || [Command entered:  sfdfd]
 || Invalid command format: sfdfd 
diff --git a/test/runtests.sh b/test/runtests.sh
index 8697679..fe60b06 100755
--- a/test/runtests.sh
+++ b/test/runtests.sh
@@ -16,6 +16,9 @@ javac  ../src/seedu/addressbook/AddressBook.java -d ../bin
 java -classpath ../bin seedu.addressbook.AddressBook ' ' < /dev/null > actual.txt
 java -classpath ../bin seedu.addressbook.AddressBook 'directoryThatDoesNotExist/valid.filename' < /dev/null >> actual.txt
 java -classpath ../bin seedu.addressbook.AddressBook '.noFilename' < /dev/null >> actual.txt
+mkdir -p data
+touch data/addressbook.txt
+java -classpath ../bin seedu.addressbook.AddressBook 'data/addressbook.txt' <exitinput.txt >>actual.txt
 java -classpath ../bin seedu.addressbook.AddressBook < input.txt >> actual.txt
 
 # compare the output to the expected output
```

But I don't feel strongly about this. I will be totally fine with it if you say "it's a waste of my time to be dealing with all of these shell scripting stuff" because I recognise that shell scripting is quite clumsy and not really the most ideal tool for these kind of tests :-)